lib/Doctrine/DBAL/Schema/OracleSchemaManager.php

```diff
@@ -36,6 +36,52 @@ class OracleSchemaManager extends AbstractSchemaManager
     /**
      * {@inheritdoc}
      */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        $tableNames = $this->listTableNames();
+
+        // Get all column definitions in one database call.
+        $allColumns = $this->_conn->fetchAll($this->_platform->getListTableColumnsSQL(null, $currentDatabase));
+
+        // Get all foreign keys definitions in one database call.
+        $allForeignKeys = $this->_conn->fetchAll($this->_platform->getListForeignKeysSQL(null, $currentDatabase));
+
+        // Get all indexes definitions in one database call.
+        $allIndexes = $this->_conn->fetchAll($this->_platform->getListTableIndexesSQL(null, $currentDatabase));
+
+        $tables = [];
+        foreach ($tableNames as $tableName) {
+            $unquotedTableName = rtrim(ltrim($tableName, '"'), '"');
```

Comment:
We need both unquoted table names (to fetch information from arrays returned by the `getListWhatever()` methods) and quoted table names that are used to create DBAL's objects names. So yes we could avoid the trim here by fetchAll from db, but we would need to re-quote the identifiers later. Since `listTableNames()` does all the checks/validations upfront, I'd prefer to keep this as is.